### PR TITLE
Correct handle negative model versions

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/model_service.cpp [hunk 1]
```diff
@@ -54,8 +54,8 @@ Status GetModelStatusImpl::createGrpcRequest(std::string model_name, const std::
     request->mutable_model_spec()->set_name(model_name);
     if (model_version.has_value()) {
         if (model_version.value() < 0) {
-            spdlog::error("Version in GetModelStatus request cannot be negative. Provided value {}", model_version.value());
-            return StatusCode::MODEL_VERSION_MISSING;
+            SPDLOG_ERROR("Illegal model version requested:{}. Model version should be a number no less than 0.", model_version.value());
+            return StatusCode::MODEL_VERSION_ILLEGAL_VALUE;
         }
         request->mutable_model_spec()->mutable_version()->set_value(model_version.value());
```
Comment (dkalinowski) [inline src/model_service.cpp:60]: What if user requests version `-5` in `GetModelStatusImpl::getModelStatus`?

File: src/test/prediction_service_utils_test.cpp [hunk 2]
```diff
@@ -78,31 +78,31 @@ class ModelWithModelInstanceFakeLoad : public ovms::Model {
 
 std::shared_ptr<ModelWithModelInstanceFakeLoad> modelWithModelInstanceFakeLoad;
 
-TEST_F(GetModelInstanceTest, WithRequested0VersionUnloadedShouldReturnModelNotLoadedAnymore) {
+TEST_F(GetModelInstanceTest, WithRequestedDefaultVersionUnloadedShouldReturnNoModelVersionAvailable) {
```
Comment (dkalinowski) [inline src/test/prediction_service_utils_test.cpp:81]: Test specifies that NO_MODEL_VERSION_AVAILABLE will be returned but it checks if status is MODEL_VERSION_MISSING.

File: src/model_service.cpp (hunk 1 above)
Comment (michalkulakowski) [reply src/model_service.cpp:60]: getModelStatus will return MODEL_VERSION_ILLEGAL_VALUE(see GetModelMetadataImpl::validate)

Comment (dkalinowski) [reply src/model_service.cpp:60]: This is GetModelStatus, not GetModelMetadata. These are 2 separate endpoints. GetModelStatus does not perform validation.

Comment (dkalinowski) [reply src/model_service.cpp:60]: Maybe add such unit test to ensure MODEL_VERSION_ILLEGAL_VALUE will be returned?

File: src/modelconfig.cpp [hunk 3]
```diff
@@ -318,8 +318,9 @@ Status ModelConfig::parseNode(const rapidjson::Value& v) {
     }
     if (v.HasMember("target_device"))
         this->setTargetDevice(v["target_device"].GetString());
-    if (v.HasMember("version"))
-        this->setVersion(v["version"].GetInt64());
+    if (v.HasMember("version")) {
+        this->setVersion(v["version"].GetUint64());
```
Comment (atobiszei) [inline src/modelconfig.cpp:322]: What will happen in case use does pass -1/0 here?

**Comment by atobiszei** [comment]
May be worthwhile to add tests for trying to add version 0 to ensure this will not be broken/is behaving as expected.

**Review by atobiszei — COMMENTED** [review]
General comment:
@dkalinowski / @michalkulakowski 
Do we want to differentiate behavior in case user request negative version in predictRequest/modelstatus/modelmetadata?

I am not sure what is python version behavior here (@michalkulakowski ?) but I am wondering if MODEL_VERSION_MISSING is still fine -> we are not serving negative versions so if user request one it is normal to reposnd with MODEL_VERSION_MISSIGN.

@michalkulakowski I don't see the ovms -> grpc status translation. In case we go with adding new retcode we should handle that.

**Review by atobiszei — APPROVED** [review]

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**